qa-tests-backend/src/main/groovy/util/SplunkUtil.groovy

```diff
@@ -141,6 +133,26 @@ class SplunkUtil {
         }
     }
 
+    static String createHECToken(int port, String tokenName = "stackrox") {
+        Response response = null
+        withRetry(20, 15) {
+            response =  given().auth()
+                .basic("admin", SPLUNK_ADMIN_PASSWORD)
+                .formParam("name", tokenName)
+                .formParam("output_mode", "json")
+                .post("https://127.0.0.1:${port}/servicesNS/nobody/search/data/inputs/http")
+        }
+        return unmarshalHEC(response?.asString())
+    }
```

Comment:
```suggestion
    static String createHECToken(int port, String tokenName = "stackrox") {
        Response response = null
        withRetry(20, 15) {
            log.info("Attempting to create new HEC ingest token")
            response =  given().auth()
                .basic("admin", SPLUNK_ADMIN_PASSWORD)
                .formParam("name", tokenName)
                .formParam("output_mode", "json")
                .post("https://127.0.0.1:${port}/servicesNS/nobody/search/data/inputs/http")
        }
        return unmarshalHEC(response?.asString())
    }
```


This would make the logs a bit easier to understand. Currently they print:

```
    15:02:54 | INFO  | IntegrationsTest          | SplunkNotifier            | Creating HEC ingest token
    15:02:54 | DEBUG | IntegrationsTest          | Helpers                   | Caught exception. Retrying in 15s (attempt 0 of 20): javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
    15:03:09 | DEBUG | IntegrationsTest          | Helpers                   | Caught exception. Retrying in 15s (attempt 1 of 20): javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
    15:03:24 | DEBUG | IntegrationsTest          | Helpers                   | Caught exception. Retrying in 15s (attempt 2 of 20): javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
    15:03:39 | DEBUG | IntegrationsTest          | Helpers                   | Caught exception. Retrying in 15s (attempt 3 of 20): javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
    15:03:54 | DEBUG | IntegrationsTest          | Helpers                   | Caught exception. Retrying in 15s (attempt 4 of 20): javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
    15:04:10 | DEBUG | IntegrationsTest          | Helpers                   | Caught exception. Retrying in 15s (attempt 5 of 20): javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
    15:04:27 | DEBUG | IntegrationsTest          | SplunkNotifier            | validating splunk deployment is ready to accept events before creating notifier...
```

I think, it would be cool to know that the handshake errors are coming from the  `createHECToken` function.

Reply:
Note that after the `validating splunk deployment is ready to accept events before creating notifier...` there will be another group of similar errors repeated only 6 times - it would be cool to add more logging about where those errors are originating from.

For `legacy=true`, we see:
```
    15:07:47 | DEBUG | IntegrationsTest          | SplunkNotifier            | validating splunk deployment is ready to accept events before creating notifier...
    15:07:48 | DEBUG | IntegrationsTest          | Helpers                   | Caught exception. Retrying in 15s (attempt 0 of 6): javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
    15:08:03 | DEBUG | IntegrationsTest          | Helpers                   | Caught exception. Retrying in 15s (attempt 1 of 6): javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
    15:08:18 | DEBUG | IntegrationsTest          | Helpers                   | Caught exception. Retrying in 15s (attempt 2 of 6): javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake
    15:08:34 | DEBUG | IntegrationsTest          | SplunkUtil                | {"sid":"1721747314.1"}
    15:08:34 | DEBUG | IntegrationsTest          | SplunkUtil                | New Search created: 1721747314.1
```

My questions here are:
- Is 6 repetitions enough for legacy=true assuming that legacy=false sometimes needs 11 out of planned 30?
- Can we maybe wait separately for Splunk to be ready and accept the handshake and then do the bussinnes logic (i.e, create token or creating notifier)?

Reply:
Reading this test again, I think it makes more sense to have a `waitForSplunkBoot` (or something similar) that is called after the container creation. I've already identified a [good API candidate for that](https://docs.splunk.com/Documentation/Splunk/7.3.1/RESTREF/RESTintrospect#server.2Fhealth.2Fsplunkd).
That way, we could have clean logs and do away with this workaround of running a `createSearch` just to ensure that Splunk is ready.
That should also result in these API calls not erroring out with an exception multiple times.

Reply:
Do you think, you could add the `waitForSplunkBoot` to this PR? Or is it too much for now?

Reply:
Yes, I added this as utility function in [02e341c](https://github.com/stackrox/stackrox/pull/12090/commits/02e341c86271641fed5871bb4009c8926a1f452d).
I will follow up with a separate PR to update the the other Splunk groovy test (SplunkIntegrationTest) with this new function.

Reply:
Okay, but let's please not postpone this follow-up too much if possible. I wish, we do our best on the Splunk test and hope to not see them flake for a long time.